tests/unittests/utils/tabular/features/conftest.py

```diff
@@ -0,0 +1,182 @@
+                # Can't call fit_transform with duplicate column names
+                generator.fit_transform(input_data_with_duplicate_columns)
+
+        assert not generator.is_fit()
+        output_data = generator.fit_transform(input_data)
+        assert generator.is_fit()
+        with pytest.raises(AssertionError):
+            # Can't call fit_transform after fit
+            generator.fit_transform(input_data)
+    
+        # Ensure input_data is not altered inplace
+        assert input_data.equals(original_input_data)
+
+        # Ensure unchanged row count
+        assert len(input_data) == len(output_data)
+
+        # Ensure transform and fit_transform output are the same for training data
+        output_data_transform = generator.transform(input_data)
+        assert output_data.equals(output_data_transform)
+
+        # Ensure transform will be the same if unnecessary columns are removed from input
+        output_data_transform = generator.transform(input_data[generator.features_in])
+        assert output_data.equals(output_data_transform)
+
+        # Ensure transform will be the same if input feature order is not the same as generator.features_in
+        reversed_column_names = list(input_data.columns)
+        reversed_column_names.reverse()
+        output_data_transform = generator.transform(input_data[reversed_column_names])
+        assert output_data.equals(output_data_transform)
+
+        # Ensure output feature order is correct
+        assert generator.features_out == list(output_data.columns)
+
+        if generator.features_in:
+            with pytest.raises(KeyError):
+                # Error if missing input feature
+                generator.transform(input_data.drop(generator.features_in[0]))
+            with pytest.raises(KeyError):
+                # Error if missing all input features
+                generator.transform(pd.DataFrame())
+
+        # Ensure unknown input columns don't affect output
+        input_data_with_extra = copy.deepcopy(input_data)
+        input_data_with_extra['__UNKNOWN_COLUMN__'] = 0
+        output_data_transform = generator.transform(input_data_with_extra)
+        assert output_data.equals(output_data_transform)
+
+        # Ensure feature_metadata_in is as expected
+        if expected_feature_metadata_in_full is not None:
+            assert expected_feature_metadata_in_full == generator.feature_metadata_in.get_feature_metadata_full()
+        # Ensure feature_metadata is as expected
+        if expected_feature_metadata_full is not None:
+            assert expected_feature_metadata_full == generator.feature_metadata.get_feature_metadata_full()
+    
+        return output_data
+
+
+class DataHelper:
+    @staticmethod
+    def generate_obj_feature() -> Series:
+        return Series(['a', 'b', 'a', 'd', 'd', 'd', 'c', np.nan, np.nan])
+
+    @staticmethod
+    def generate_int_feature() -> Series:
```

Comment:
Add a large int value here?

Reply:
I think that can be part of a separate dedicated feature check for generators that are sensitive to value size. Generally I think we should trust pandas/numpy to not do weird things with large ints.